src/transaction.cpp

```diff
@@ -1073,9 +1089,7 @@ bool wsrep::transaction::bf_abort(
                 }
             }
 
-            lock.unlock();
-            server_service_.background_rollback(client_state_);
-            lock.lock();
+            server_service_.background_rollback(lock, client_state_);
```

Comment:
1) client_state exposes mutex(), which makes this change is unnecessary. But I'm not really against this... just something to consider.
2)  Implementation of background_rollback() simply enqueues transaction for rollbacker thread... and this is the very last thing we do in bf_abort() before returning and releasing the lock anyway. Not sure what race condition is avoided here.